week-1-ts-ecosystem/w01e02-enter-the-world-of-typescript/src/shop/hero.component.ts

```diff
@@ -0,0 +1,18 @@
+import { div, p, section } from '../framework/dom-creators';
+
+type Hero = {
+  title: string;
+  subTitle: string;
+};
```

Comment:
> Czy nie przeszkadza to, że typy blisko związane z danym komponentem są w tym samym pliku co komponent?
> ~ @tars-mj 

Oczywiście że nie przeszkadza. I moim zdaniem to dobra praktyka.
Nie używasz nigdzie więcej typu `Hero` jest on Ci potrzebny tylko ze względu na typowanie `props` dla komponentu.

Powiem więcej, ja taki typ po prostu nazwałbym `PropTypes` w każdym komponencie nie zastanawiając się nad nadawaniem mu finezyjnych nazw.

Idea jest następująca. Skoro mamy kod taki:
```
function hero({ title, subTitle }) {....
```

To _TypeScript_ "wymusza" na nas typowanie tych parametrów. I to jest dobre.
Jednak z perspektywy projektu, najczęściej będzie to typ, który użyjemy tylko w tym komponencie.
Nawet jeśli do komponentu przekazujemy coś z modelu danych - to najczęściej nie będzie to `1:1` model
bardzo często będzie to więcej `props` niż tylko sam `model` 

Dlatego też tutaj jest miejsce na pokazanie "Zobacz ten komponent potrzebuje tackich props"
i to tutaj możemy też łączyć typy - jeśli jest taka potrzeba.